code/API_definitions/qos-booking-and-assignment.yaml

```
@@ -0,0 +1,1563 @@
+        required: true
+      responses:
+        "200":
+          description: Returns the QoS booking information for a given device. A device may have multiple bookings (for several times and locations), thus the response is an array. An empty array is returned if no sessions are found.
+          headers:
+            x-correlator:
+              $ref: "#/components/headers/x-correlator"
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/RetrieveBookingsOutput"
+        "400":
+          $ref: "#/components/responses/Generic400"
+        "401":
+          $ref: "#/components/responses/Generic401"
+        "403":
+          $ref: "#/components/responses/Generic403"
+        "404":
+          $ref: "#/components/responses/GenericDevice404"
+        "422":
+          $ref: "#/components/responses/Generic422"
+        "429":
+          $ref: "#/components/responses/Generic429"
+      security:
+        - openId:
+            - "qos-booking-and-assignment:qos-bookings:retrieve-by-device"
+
+components:
+  securitySchemes:
+    openId:
+      description: OpenID Connect authentication
+      type: openIdConnect
+      openIdConnectUrl: https://example.com/.well-known/openid-configuration
+    notificationsBearerAuth:
+      description: Bearer authentication for notifications
+      type: http
+      scheme: bearer
+      bearerFormat: "{$request.body#/sinkCredential.credentialType}"
+
+  parameters:
+    BookingId:
+      name: bookingId
+      in: path
+      description: Booking ID that was obtained from the createBooking operation
+      required: true
+      schema:
+        $ref: "#/components/schemas/BookingId"
+
+    x-correlator:
+      name: x-correlator
+      in: header
+      description: Correlation id for the different services
+      schema:
+        $ref: "#/components/schemas/XCorrelator"
+
+
+  headers:
+    x-correlator:
+      description: Correlation id for the different services
+      schema:
+        $ref: "#/components/schemas/XCorrelator"
+
+  schemas:
+    BookingId:
+      description: Booking Identifier in UUID format
+      type: string
+      format: uuid
+    
+    BookingInput: 
+      description: |
+        Basic input that are required for to book QoS in advance.  
+        - 'numDevices' is optional and it is defaulted to 1.  
+        - 'numDevices' represents the maximum number of devices the enduser wants the operator to reserve
+      allOf:
+        - type: object
+          properties:
+            numDevices:
+              $ref: "#/components/schemas/DeviceCount"                   
+            qosProfile:
+              $ref: "#/components/schemas/QosProfileName"
+            startTime:
+              $ref: "#/components/schemas/Time"
+            duration:
+              $ref: "#/components/schemas/Duration"
+            serviceArea:
+              $ref: "#/components/schemas/Area"   

```

Trailing spaces (after M3 we should enable linting in the repository to check both APIs):

```suggestion
              $ref: "#/components/schemas/Area"
```